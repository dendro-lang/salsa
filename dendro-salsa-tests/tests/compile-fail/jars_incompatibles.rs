#[salsa::jar(db = Db, return_ref)]
struct JarWithRetRef(MyInput);

#[salsa::jar(db = Db, specify)]
struct JarWithDb(MyInput);


#[salsa::jar(db = Db, no_eq)]
struct JarWithNoEq(MyInput);

#[salsa::jar(db = Db, jar = Jar)]
struct JarWithJar(MyInput);

#[salsa::jar(db = Db, data = Data)]
struct JarWithData(MyInput);

#[salsa::jar(db = Db, recovery_fn = recover)]
struct JarWithRecover(MyInput);

#[salsa::jar(db = Db, lru = 32)]
struct JarWithLru(MyInput);

#[salsa::jar(db = Db, constructor = JarConstructor)]
struct JarWithConstructor(MyInput);

#[salsa::input(jar = Jar1)]
struct MyInput {
    field: u32,
}

fn main() {

}