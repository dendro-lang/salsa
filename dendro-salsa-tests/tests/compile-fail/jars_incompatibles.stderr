error: `return_ref` option not allowed here
 --> tests/compile-fail/jars_incompatibles.rs:1:23
  |
1 | #[salsa::jar(db = Db, return_ref)]
  |                       ^^^^^^^^^^

error: `specify` option not allowed here
 --> tests/compile-fail/jars_incompatibles.rs:4:23
  |
4 | #[salsa::jar(db = Db, specify)]
  |                       ^^^^^^^

error: `no_eq` option not allowed here
 --> tests/compile-fail/jars_incompatibles.rs:8:23
  |
8 | #[salsa::jar(db = Db, no_eq)]
  |                       ^^^^^

error: `jar` option not allowed here
  --> tests/compile-fail/jars_incompatibles.rs:11:23
   |
11 | #[salsa::jar(db = Db, jar = Jar)]
   |                       ^^^

error: `data` option not allowed here
  --> tests/compile-fail/jars_incompatibles.rs:14:23
   |
14 | #[salsa::jar(db = Db, data = Data)]
   |                       ^^^^

error: `recovery_fn` option not allowed here
  --> tests/compile-fail/jars_incompatibles.rs:17:23
   |
17 | #[salsa::jar(db = Db, recovery_fn = recover)]
   |                       ^^^^^^^^^^^

error: `lru` option not allowed here
  --> tests/compile-fail/jars_incompatibles.rs:20:23
   |
20 | #[salsa::jar(db = Db, lru = 32)]
   |                       ^^^

error: `constructor` option not allowed here
  --> tests/compile-fail/jars_incompatibles.rs:23:23
   |
23 | #[salsa::jar(db = Db, constructor = JarConstructor)]
   |                       ^^^^^^^^^^^

error[E0412]: cannot find type `Jar1` in this scope
  --> tests/compile-fail/jars_incompatibles.rs:26:22
   |
26 | #[salsa::input(jar = Jar1)]
   |                      ^^^^ not found in this scope

error[E0412]: cannot find type `Jar1` in this scope
  --> tests/compile-fail/jars_incompatibles.rs:26:22
   |
26 | #[salsa::input(jar = Jar1)]
   |                      ^^^^ not found in this scope
   |
help: you might be missing a type parameter
   |
27 | struct MyInput<Jar1> {
   |               ++++++
